planning/behavior_path_planner/src/occupancy_grid_map/occupancy_grid_map.cpp

```diff
@@ -0,0 +1,205 @@
+// Copyright 2021 Tier IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "behavior_path_planner/occupancy_grid_map/occupancy_grid_map.hpp"
+
+#include <tier4_autoware_utils/geometry/geometry.hpp>
+#include <tier4_autoware_utils/tier4_autoware_utils.hpp>
+
+#include <vector>
+
+namespace behavior_path_planner
+{
+using tier4_autoware_utils::createQuaternionFromYaw;
+using tier4_autoware_utils::normalizeRadian;
+using tier4_autoware_utils::transformPose;
+
+int discretizeAngle(const double theta, const int theta_size)
+{
+  const double one_angle_range = 2.0 * M_PI / theta_size;
+  return static_cast<int>(std::rint(normalizeRadian(theta, 0.0) / one_angle_range)) % theta_size;
+}
+
+IndexXYT pose2index(
+  const nav_msgs::msg::OccupancyGrid & costmap, const geometry_msgs::msg::Pose & pose_local,
+  const int theta_size)
+{
+  const int index_x = pose_local.position.x / costmap.info.resolution;
+  const int index_y = pose_local.position.y / costmap.info.resolution;
+  const int index_theta = discretizeAngle(tf2::getYaw(pose_local.orientation), theta_size);
+  return {index_x, index_y, index_theta};
+}
+
+geometry_msgs::msg::Pose index2pose(
+  const nav_msgs::msg::OccupancyGrid & costmap, const IndexXYT & index, const int theta_size)
+{
+  geometry_msgs::msg::Pose pose_local;
+
+  pose_local.position.x = index.x * costmap.info.resolution;
+  pose_local.position.y = index.y * costmap.info.resolution;
+
+  const double one_angle_range = 2.0 * M_PI / theta_size;
+  const double yaw = index.theta * one_angle_range;
+  pose_local.orientation = createQuaternionFromYaw(yaw);
+
+  return pose_local;
+}
+
+geometry_msgs::msg::Pose global2local(
```

Comment:
please consider using free space planner's package 
https://github.com/autowarefoundation/autoware.universe/blob/1bab84e75e6a6b750b5781212b04702b7646616f/planning/freespace_planning_algorithms/src/abstract_algorithm.cpp#L35
![image](https://user-images.githubusercontent.com/65527974/177666612-65c698d3-e361-4274-9add-77349b2fdcd3.png)

or make library later to reduce maintenance cost.

Reply:
Yes, I want to make it common since it is almost the same. 
Currently, It can't include free_space_planner in behavior_path, so it is like a duplicate. I would like to leave it as it is this time...

Reply:
Then Please add comment TODO that this is duplicated and will be replaced to common package